api/img_streamer.go

```diff
@@ -0,0 +1,112 @@
+package api
+
+import (
+	"encoding/binary"
+	"fmt"
+	img_streamer "github.com/cedana/cedana/api/services/img_streamer"
+	"google.golang.org/protobuf/proto"
+	"net"
+	"os/signal"
+	"path/filepath"
+	"sync"
+	"syscall"
+)
+
+const (
+	IMG_STREAMER_CAPTURE_SOCKET_NAME = "ced-capture.sock"
+	IMG_STREAMER_SERVE_SOCKET_NAME   = "ced-serve.sock"
+	O_DUMP                           = 577
+	O_RSTR                           = 578
+)
+
+var (
+	imgStreamerFdLock sync.Mutex
+	imgStreamerMode   int
+)
+
+func socketNameForMode(mode int) string {
+	switch mode {
+	case O_DUMP:
+		return IMG_STREAMER_CAPTURE_SOCKET_NAME
+	case O_RSTR:
+		return IMG_STREAMER_SERVE_SOCKET_NAME
+	default:
+		panic("BUG")
+	}
+}
+
+func imgStreamerInit(imageDir string, mode int) (net.Conn, error) {
```

Comment:
I see that we convert it to `*net.UnixConn` for usage later. We can do the conversion within this function, and return the explicit type.

This allows us to avoid possible errors within the functions later.